### ISSUE TITLE
Wrong generation of "consumes" and "produces" fields of swagger.json file

### ISSUE THREAD

**Issue by fpotera** [stoicflame/enunciate#628]
Using enunciate 2.9.0 version I get the following fields in swagger.json for all rest services:

        "consumes" : [ "com.webcohesion.enunciate.modules.jaxrs.api.impl.MediaTypeDescriptorImpl@54bb6cee", "com.webcohesion.enunciate.modules.jaxrs.api.impl.MediaTypeDescriptorImpl@7cad6681" ],
        "produces" : [ "com.webcohesion.enunciate.modules.jaxrs.api.impl.MediaTypeDescriptorImpl@67859020" ],

instead of:

        "consumes" : [ "application/json", "application/xml" ],
        "produces" : [ "application/json" ],

I think, this is not a normal behavior.
Thank you.

**Comment by stoicflame** [comment]
Huh. That is really baffling.

I'm afraid I don't see what you're reporting, and I have no idea how that's even possible. A run of all the [examples](https://github.com/stoicflame/enunciate/tree/master/examples) and the [getting started sample](https://github.com/stoicflame/enunciate-sample) doesn't reveal the issue.

Can you shed any light on what you might be doing differently?

**Comment by spidersouleater** [comment]
I have the same issue. The bug seems to appear with the version 2.7.0

**Comment by stoicflame** [comment]
The only explanation I have is that you're using an old/outdated version of the `swagger.fmt` template file that Enunciate uses.

Do you have some kind of custom swagger generation template? When you scan your classpath for `swagger.fmt`, what do you find?

**Comment by spidersouleater** [comment]
Yes, i have customized my swagger.fmt (Swagger UI) and i m using an old/outdated version. I will try it with the new version and report the result the next day.

**Comment by fpotera** [comment]
I also use the old, customized swagger.fmt file.

**Comment by spidersouleater** [comment]
When i m using not a specified freemarkerProcessingTemplate (swagger.fmt) the problem is solved.

With the freemarkerProcessingTemplate from [link](https://github.com/stoicflame/enunciate/blob/master/swagger/src/main/resources/com/webcohesion/enunciate/modules/swagger/swagger.fmt) i get the following error. I havn’t customized this file. 

```
Mai 03, 2017 9:08:03 AM freemarker.log._JULLoggerFactory$JULLogger error
SCHWERWIEGEND: Error executing FreeMarker template
FreeMarker template error:
The following has evaluated to null or missing:
==> dataFormatNameFor  [in template "…./service/swagger-ui/swagger.fmt" at line 340, column 12]

Tip: If the failing expression is known to be legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-missing[/#if
]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??

FTL stack trace ("~" means nesting-related):
        - Failed at: #elseif dataFormatNameFor(dataType)??  [in template "…./service/swagger-ui/swagger.fmt" in macro "referenceDataType" at line 340, column 3]
        - Reached through: @referenceDataType dataType=property...  [in template "…./ service/swagger-ui/swagger.fmt" at line 142, column 13]
        - Reached through: @file name="swagger.json" charset="ut...  [in template "…./ service/swagger-ui/swagger.fmt" at line 30, column 1]

Java stack trace (for programmers):
freemarker.core.InvalidReferenceException: [... Exception message was already printed; see it above ...]
        at freemarker.core.InvalidReferenceException.getInstance(InvalidReferenceException.java:131)
        at freemarker.core.UnexpectedTypeException.newDesciptionBuilder(UnexpectedTypeException.java:77)
        at freemarker.core.UnexpectedTypeException.<init>(UnexpectedTypeException.java:40)
        at freemarker.core.NonMethodException.<init>(NonMethodException.java:46)
        at freemarker.core.MethodCall._eval(MethodCall.java:84)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.ExistsExpression._eval(ExistsExpression.java:45)
        at freemarker.core.Expression.eval(Expression.java:78)
        at freemarker.core.Expression.evalToBoolean(Expression.java:124)
        at freemarker.core.Expression.evalToBoolean(Expression.java:110)
        at freemarker.core.IfBlock.accept(IfBlock.java:46)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.Macro$Context.runMacro(Macro.java:184)
        at freemarker.core.Environment.invoke(Environment.java:701)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:84)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlockInner(IteratorBlock.java:268)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlock(IteratorBlock.java:220)
        at freemarker.core.IteratorBlock$IterationContext.accept(IteratorBlock.java:194)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:572)
        at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:78)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:64)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
        at freemarker.core.IfBlock.accept(IfBlock.java:48)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlockInner(IteratorBlock.java:268)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlock(IteratorBlock.java:220)
        at freemarker.core.IteratorBlock$IterationContext.accept(IteratorBlock.java:194)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:572)
        at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:78)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:64)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlockInner(IteratorBlock.java:268)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlock(IteratorBlock.java:220)
        at freemarker.core.IteratorBlock$IterationContext.accept(IteratorBlock.java:194)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:572)
        at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:78)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:64)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visitByHiddingParent(Environment.java:345)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlockInner(IteratorBlock.java:268)
        at freemarker.core.IteratorBlock$IterationContext.executeNestedBlock(IteratorBlock.java:220)
        at freemarker.core.IteratorBlock$IterationContext.accept(IteratorBlock.java:194)
        at freemarker.core.Environment.visitIteratorBlock(Environment.java:572)
        at freemarker.core.IteratorBlock.acceptWithResult(IteratorBlock.java:78)
        at freemarker.core.IteratorBlock.accept(IteratorBlock.java:64)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.Environment$NestedElementTemplateDirectiveBody.render(Environment.java:2255)
        at com.webcohesion.enunciate.util.freemarker.FileDirective.execute(FileDirective.java:78)
        at freemarker.core.Environment.visit(Environment.java:389)
        at freemarker.core.UnifiedCall.accept(UnifiedCall.java:104)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.MixedContent.accept(MixedContent.java:54)
        at freemarker.core.Environment.visit(Environment.java:324)
        at freemarker.core.Environment.process(Environment.java:302)
        at freemarker.template.Template.process(Template.java:325)
        at com.webcohesion.enunciate.modules.swagger.SwaggerDeploymentModule.processTemplate(SwaggerDeploymentModule.java:297)
        at com.webcohesion.enunciate.modules.swagger.SwaggerDeploymentModule$SwaggerInterfaceDescription.writeTo(SwaggerDeploymentModule.java:187)
        at com.webcohesion.enunciate.modules.docs.DocsModule.call(DocsModule.java:303)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:46)
        at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:25)
        at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:80)
        at rx.internal.producers.SingleProducer.request(SingleProducer.java:65)
        at rx.Subscriber.setProducer(Subscriber.java:209)
        at rx.Subscriber.setProducer(Subscriber.java:205)
        at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:111)
        at rx.internal.operators.OperatorTakeLastOne$ParentSubscriber.emit(OperatorTakeLastOne.java:175)
        at rx.internal.operators.OperatorTakeLastOne$ParentSubscriber.onCompleted(OperatorTakeLastOne.java:141)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitLoop(OperatorMerge.java:648)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emit(OperatorMerge.java:560)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onCompleted(OperatorMerge.java:281)
        at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.fastPath(OnSubscribeFromArray.java:82)
        at rx.internal.operators.OnSubscribeFromArray$FromArrayProducer.request(OnSubscribeFromArray.java:58)
        at rx.Subscriber.setProducer(Subscriber.java:211)
        at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:32)
        at rx.internal.operators.OnSubscribeFromArray.call(OnSubscribeFromArray.java:24)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:8460)
        at rx.internal.operators.CachedObservable$CacheState.connect(CachedObservable.java:183)
        at rx.internal.operators.CachedObservable$CachedSubscribe.call(CachedObservable.java:248)
        at rx.internal.operators.CachedObservable$CachedSubscribe.call(CachedObservable.java:230)
        at rx.Observable.unsafeSubscribe(Observable.java:8460)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:248)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:145)
        at rx.internal.operators.OnSubscribeFromIterable$IterableProducer.fastpath(OnSubscribeFromIterable.java:172)
        at rx.internal.operators.OnSubscribeFromIterable$IterableProducer.request(OnSubscribeFromIterable.java:85)
        at rx.Subscriber.setProducer(Subscriber.java:211)
        at rx.internal.operators.OnSubscribeFromIterable.call(OnSubscribeFromIterable.java:62)
        at rx.internal.operators.OnSubscribeFromIterable.call(OnSubscribeFromIterable.java:33)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:50)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.subscribe(Observable.java:8553)
        at rx.Observable.subscribe(Observable.java:8520)
        at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:433)
        at rx.observables.BlockingObservable.single(BlockingObservable.java:332)
        at com.webcohesion.enunciate.EnunciateAnnotationProcessor.process(EnunciateAnnotationProcessor.java:122)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
        at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
        at com.sun.tools.javac.main.Main.compile(Main.java:523)
        at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
        at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
        at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:674)
        at com.webcohesion.enunciate.mojo.ConfigMojo.execute(ConfigMojo.java:401)
        at com.webcohesion.enunciate.mojo.AssembleBaseMojo.execute(AssembleBaseMojo.java:101)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.647s
[INFO] Finished at: Wed May 03 09:08:03 CEST 2017
[INFO] Final Memory: 49M/247M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.9.1:assemble (assemble) on project service: freemarker.core.InvalidReferenceException: The following has evaluated to null or missing:
[ERROR] ==> dataFormatNameFor  [in template "…../service/swagger-ui/swagger.fmt" at line 340, column 12]
[ERROR]
[ERROR] ----
[ERROR] Tip: If the failing expression is known to be legally refer to something that's sometimes null or missing, either specify a default value like myOptionalVar!myDefault, or use [#if myOptionalVar??]when-present[#else]when-miss
ing[/#if]. (These only cover the last step of the expression; to cover the whole expression, use parenthesis: (myOptionalVar.foo)!myDefault, (myOptionalVar.foo)??
[ERROR] ----
[ERROR]
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR] - Failed at: #elseif dataFormatNameFor(dataType)??  [in template "…../service/swagger-ui/swagger.fmt" in macro "referenceDataType" at line 340, column 3]
[ERROR] - Reached through: @referenceDataType dataType=property...  [in template "…../service service/swagger-ui/swagger.fmt" at line 142, column 13]
[ERROR] - Reached through: @file name="swagger.json" charset="ut...  [in template "…../service /swagger-ui/swagger.fmt" at line 30, column 1]
[ERROR] ----
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

I have used this option in the enunciate.xml:

`<modules>
 <swagger disabled="false" base="swagger-ui" freemarkerProcessingTemplate="swagger-ui/swagger.fmt" />
</modules>`

**Comment by stoicflame** [comment]
Okay, I'll have to look closer to figure out why that happens.

**Comment by stoicflame** [comment]
I was able to reproduce this with 2.9.1, but it seems to work with 2.10.0-SNAPSHOT, so I'm going to assume this will be resolved when 2.10 is released.

**Comment by stoicflame** [comment]
[Enunciate 2.10 is now available](https://github.com/stoicflame/enunciate/releases/tag/v2.10.0).
